zio-config/src/main/scala/zio/config/actions/ConfigDocs.scala

```diff
@@ -6,79 +6,82 @@ sealed trait ConfigDocs
 
 object ConfigDocs {
 
-  final case class Empty()                                  extends ConfigDocs
-  final case class Leaf(value: KeyDescription)              extends ConfigDocs
-  final case class And(left: ConfigDocs, right: ConfigDocs) extends ConfigDocs
-  final case class Or(left: ConfigDocs, right: ConfigDocs)  extends ConfigDocs
-
-  final case class KeyDescription(path: String, value: Option[String], docs: List[String])
+  final case class Empty()                                                                    extends ConfigDocs
+  final case class PathDetails(path: List[String], value: Option[String], docs: List[String]) extends ConfigDocs
```

Comment:
Could you use `Vector` instead of `List` to take advantage of the optimization of `++` in `Vector` (in case you're using it)